Mathlib/Algebra/GroupWithZero/NonZeroDivisors.lean

```diff
@@ -139,6 +139,16 @@ theorem nonZeroDivisors.ne_zero (hx : x ∈ M₀⁰) : x ≠ 0 :=
 @[simp]
 theorem nonZeroDivisors.coe_ne_zero (x : M₀⁰) : (x : M₀) ≠ 0 := nonZeroDivisors.ne_zero x.2
 
+instance [IsLeftCancelMulZero M₀] :
+    LeftCancelMonoid M₀⁰ where
+  mul_left_cancel _ _ _ h :=  Subtype.ext <|
+    mul_left_cancel₀ (nonZeroDivisors.coe_ne_zero _) (by simpa [Subtype.ext_iff] using h)
```

Comment:
Do you understand why these 2 instances make the file almost 10% slower? Are they slow, or do they slow down instance search elsewhere? Or the file is very short?

Reply:
The instances took about 1/10th of a second each. I have squeezed the `simp` so they are now much faster